giftless/storage/__init__.py

```diff
@@ -32,6 +32,9 @@ def exists(self, prefix: str, oid: str) -> bool:
     def get_size(self, prefix: str, oid: str) -> int:
         pass
 
+    def get_mime_type(self, prefix: str, oid: str) -> str:
```

Comment:
Might be good to have a default `return "application/octet-stream"` implementation here, this will allow storage backends to not implement this and still have reasonable behavior.